<?xml version="1.0"?>
<launch>

    <!-- <arg name="name" default="$(env SAMBOT_ID)"/> -->
    <arg name="name" default="sambot1"/>
    <arg name="local_planner" default="base" doc="Local planner can be either dwa, dwb, base, teb or pose"/>

    <group ns="$(arg name)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find sambot_navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
        <rosparam file="$(find sambot_navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find sambot_navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find sambot_navigation)/config/global_costmap_params.yaml" command="load" /> 

        <rosparam file="$(find sambot_navigation)/config/$(arg local_planner)_local_planner_params.yaml" command="load" />
            <rosparam file="$(find sambot_navigation)/config/move_base_params.yaml" command="load" />
        <!-- <rosparam file="$(find sambot_navigation)/config/base_local_planner_params.yaml" command="load" /> -->

        <param name="global_costmap/global_frame" value="map" />
        <param name="global_costmap/robot_base_frame" value="$(arg name)/chassis" />
        <param name="local_costmap/global_frame" value="$(arg name)/odom" />
        <param name="local_costmap/robot_base_frame" value="$(arg name)/chassis" />


        <remap from="map" to="/map"/>
        <remap from="odom" to="mobile_base_controller/odom"/>
        <remap from="cmd_vel" to="mobile_base_controller/cmd_vel"/>  
    </node>
    </group>

</launch>
